impl/client/src/main/javascript/web/pentaho/visual/config/vizApi.conf.js

```diff
@@ -44,6 +44,31 @@ define(function() {
 
   return {
     rules: [
+      // region Viz API Standard Color Palettes Context Registration
+      {
+        priority: RULE_PRIO_VIZ_DEFAULT,
+        select: {
+          type: "pentaho/type/Context"
+        },
+        apply: {
+          instances: {
+            "pentaho/visual/color/palettes/nominalPrimary":    {type: "pentaho/visual/color/palette", priority: -10},
```

Comment:
A totally different way of registering stuff. We have `pentaho/service`, `pentaho/typeInfo` and by Context configuration.

Reply:
Yes.

1. `pentaho/service` is a low level inversion of control mechanism for AMD modules
2. `pentaho/typeInfo` declares types and works in the realm of types, not AMD modules; it happens that, currently, the ids of types are the ids of their AMD modules. Loading types by base type, in the Context, requires only that we know the relationship between types; where a type lives is a secondary deal; in our case, by convention, the type lives in a module with the same id as the type and type modules have a known value type. The metadata about types does not change with the environment.
3. Instances are yet another different kind of thing. We could argue that an instance does not change its type depending on the environment. So, it could have an environment independent way of stating its type. The priority however is inevitably tied to the environment and is tied to the configuration of the Context.

I also remember discussing the Context configuration solution with you. The part about changing to use `pentaho/typeInfo` instead of `pentaho/service`, that was mentioned several times, as a future thing I'd like to change, but we never discussed its pros and cons.

So, I do believe different kinds of problems require different solutions. For me, this is an instance of that general rule.

Reply:
To revisit in the future the "ways of registering stuff".